Stage 1

AmarKayed opened the pull request.

AmarKayed committed:
Added Project Diagram

This diagram was made using draw.io and represents the model for implementing the project's classes.

AmarKayed committed:
Started Working On README.md

Started adding some description and context to the project's main classes and logistics.

AmarKayed committed:
Updated README.md V2

Explained the Employee, Account and CurrentAccount classes in the README.md file.

AmarKayed committed:
Updated README.md V3

Finished describing the entities. I still have some small patches to perform on the README.md file such as uploading the UML diagram of the class hierarchy or adding/removing some business logic. This commit was performed earlier today at 15:04:22 but I had to edit it and push it again.

AmarKayed committed:
Added UML Diagram In README.md V4

It took me a while to figure out why it wasn't working, apparently I was using "!()[]" instead of "![]()".

AmarKayed committed:
Initialized Java Project

AmarKayed committed:
Added To-Do List For Next Day

AmarKayed committed:
Added Person POJO Class

I've also modified the positions of Person.firstName and Person.lastName in the "Banking-Diagram.png" image file. Lastly, I've created the "add-to-readme.txt" file for notes that I should add to the "README.md" file once I finish implementing the code for the project.

AmarKayed committed:
Added Address POJO Class

I've also modified the positions of Person.firstName and Person.lastName in the Person POJO class.

AmarKayed committed:
Added Customer POJO Class

I had to first create the Address class in order to encapsulate it in the Customer class.

AmarKayed committed:
Rearranged Class Methods Order

AmarKayed committed:
Added Employee POJO Class

AmarKayed committed:
Added Account POJO Class

AmarKayed committed:
Added Transaction POJO Class

I had to start with the Transaction class because I will be needing it for the CurrentAccount class.

AmarKayed committed:
Added CurrentAccount POJO Class

I've also tested some code and created the "DraftClassIgnore.java" file for testing further unused code.

AmarKayed committed:
Added SavingsAccount POJO Class

AmarKayed committed:
Started Working On The Service Menu

This is the general service menu which will be further completed with the individual services of each class.

AmarKayed committed:
Started Working On The "Add Person" Option

Currently, I've only implemented the menu part, but I'm trying to redesign the code such that I have an interface and a service class strictly for the Person base class. The Person POJO class's service class will be called from the general "Service Menu" class whenever the "A" option will be selected. Furthermore, I will be trying to make the Person's service class call the Customer's and the Employee's service classes in order to better differntiate between which type of Person extension we're adding to our general Person ArrayList.

AmarKayed committed:
Finished First Part Of Adding Person

I still have to implement the reading of the Person object before I add it to the list tomorrow.

AmarKayed committed:
Finished Adding Employee + Added "input.txt" File

I still have to implement adding Customer as well. Furthermore, I have to focus at fixing the problem with temporary objects taking reserving ID numbers. For example, when I add Employees, their ID number starts with 3, then 6 and so on, this happens because creating an Employee implies creating a temporary Person and Employee object, which take up the ID numbers 1 and 2 before the actual Employee object that we want to create takes the ID number 3.

AmarKayed committed:
Bug Fix: ID Numbers

The last commit was slightly wrong: the description was correct but it didn't correspond to the code. In the code we were actually creating three temporary objects(2 Persons and 1 Employee). The first Person was created in PersonService.readPerson() on line 31, the second Person was created ACCIDENTALLY in EmployeeService.readEmployee() on line 25, the third person was actually an Employee created in EmployeeService.readEmployee() on line 53. The description of the last commit was correct: we really needed a temporary Person and one temporary Employee in order to create a correct Employee object. The mistake was that we created a second temporary Person object when we were trying to pull the returned temporary Person object from PersonService.readPerson(), this happened because we were actually creating a new copy of that object(using new Person()) instead of just referencing it. The fix was that we simply changed the line 25 of EmployeeService.readEmployee() such that it only references the returned object, it doesn't copy it. Thus, we will increase the ID once for the first temporary Person object and then immediately decrease it, we reference that object and create a new temporary Employee but this time we don't decrease the ID counter. I'm still undecided if we should add the reference of the temporary Employee created in EmployeeService.readEmployee() to the personList or if we should create a copy of the EmployeeService.readEmployee() returned object in PersonService.addPerson(), instead of using the reference.

AmarKayed committed:
Finished Adding Customer + Reorganized Package Folders

AmarKayed committed:
Started Working On Opening Accounts

So far I've only implemented reading base-class Account objects. In the next commits I will be implementing reading and adding CurrentAccount and SavingsAccount objects.

AmarKayed committed:
Finished Opening Savings Account

I also fixed some issues with the toString() functions in CurrentAccount and SavingsAccount. I added a couple of new options in the menu. I only have to implement the services for CurrentAccount and Transaction.

AmarKayed committed:
Finished Opening Current Account

I also finished implementing the TransactionService class.

AmarKayed committed:
Added Delete Person

I now have to implement the feature of closing accounts when deleting customer/account holder.

AmarKayed committed:
Bug Fix: Resolved Closing Accounts After Person Deletion

I also noticed a problem with the "K" command, namely that after I close the accounts of a deleted Person object, the AccountService.accountHashMap will have it's keys not directly correspond/equal the account's IBAN. Thus, when selecting a choice based on the index of the account that we want to close using the "K" command, the code will try to delete an account whose IBAN is equal to the choice index number, not the actual key of the HashMap or the IBAN of the account.

AmarKayed committed:
Added Update Person(only base-class)

So far I've only implemented the update operation for the Person class. In the next commit I will be adding this functionality to the Customer and Employee classes. Furthermore, when we update a person his/her details get updated in the accounts they posses as well.

AmarKayed committed:
Added Employee Update

In the next commit I will be implementing the Customer update.

AmarKayed committed:
Finished CRUD On Person + Customer + Employee + Address Classes

I finished the update Customer feature which implies updating Address as well.

AmarKayed committed:
Added Sort + Deposit Transactions + Withdraw Transactions

Final commit before project deadline.

diana-maftei commented on project/src/com/company/draftignore/DraftClassIgnore.java:
```diff
@@ -0,0 +1,128 @@
+package com.company.draftignore;
```
[Fix] fiindca folosim git, astfel de fisiere le putem adauga la "stash", sau le putem pune pe alt branch, nu vrem sa commitem si sa mergeuim in main (care este "production"-ul nostru) cod sau fisiere nefolositoare
https://www.atlassian.com/git/tutorials/saving-changes/git-stash

diana-maftei commented on project/src/com/company/transaction/Transaction.java:
```diff
@@ -0,0 +1,110 @@
+    private String transactionDate;
+    private float amount;
+
+    private static List<String> monthList = new ArrayList<String>(Arrays.asList("JAN", "FEB", "MAR", "APR", "MAY", "JUN", "JUL", "AUG", "SEP", "OCT", "NOV","DEC" ));
```
[Improvement] ai putea folosi un enum pentru astea, si nu vad de ce am tine in clasa Transaction aceasta informatie

AmarKayed replied:
Intr-adevar nu prea are sens sa las in Transaction aceasta informatie, insa aveam nevoie de o lista/colectie cu toate lunile accesibila, intrucat ma foloseam de ea atunci cand implementam metoda compareTo() (sortarea se facea in functie de data tranzactiei). Am ales sa pastrez aceasta lista, intrucat daca as folosi un enum/clasa ar trebui sa implementez si logica de comparatie/precedenta a lunilor(sa zic ca ianuarie e inainte de februarie, etc). 

O sa dau un citat super lazy dar cu o morala pozitiva: "If it ain't broke, don't fix it" :))))

diana-maftei commented on project/src/com/company/transaction/Transaction.java:
```diff
@@ -0,0 +1,110 @@
+
+public class Transaction implements Comparable{
+
+    private final int IBAN;
```
[Fix] variabilele in java se scriu cu pascalCase, cu litere mari se scriu doar constantele, care sunt statice si finale. nu este si cazul acesta. Stiu ca IBAN este un acronim, dar chiar si asa, respectam naming convention-urile fiindca altfel, la o prima vedere, am putea crede ca este o constanta

AmarKayed replied:
Aveti dreptate, acum observ ca mi-a scapat acest lucru din neatentie, intrucat mereu citeam IBAN cu litere mare. La restul atributelor, pachetelor, claselor si metodelor am pastrat naming convention-urile din java. 

Am doar o intrebare: atunci cand incep sa tratez aceste greseli, pot sa dau commit pe branch-ul stage-1 cum am facut pana acum sau acest lucru va afecta pull request-ul?

AmarKayed replied:
Am folosit refractor-ul din IntelliJ ca sa redenumesc fiecare aparitie a lui IBAN in iban(inclusiv in numele metodelor). Am ales totusi sa las IBAN in diagrama pentru ierarhia claselor din README.md, intrucat zic ca arata mai bine asa, este in regula sau doriti sa schimb si acolo?

diana-maftei commented on project/src/com/company/accounts/Account.java:
```diff
@@ -0,0 +1,99 @@
+package com.company.accounts;
```
[Fix] va vorbisem de 2 modalitati de a imparti clasele pe pachete, ori dupa layere, ori dupa buness logic, tu pari ca le-ai amestecat pe cele doua. Ori ai un pachet entitati (si pui toate pojourile) si services (si pui toate serviciile), ori imparti dupa business logic, adica accounts (care o sa contina si pojouri si servicii), persons (si pojo si services), etc, cam cum ai facut cu transaction si address

AmarKayed replied:
Pentru a 2-a varianta presupun ca nu pot sa am spre exemplu cate un pachet pentru fiecare entitate si practic sa am:
person(cu pojo, interfata si service strict pentru clasa Person)
employee(doar cu pojo-ul, intervata si service-ul pentru clasa Employee)

si asa mai departe, intrucat person si employee sunt legate prin mostenire, corect?

Practic ar fi mai ok sa am Person, Customer si Employee cu toate pojo-urile si serviciile lor intr-un pachet persons, Account, CurrentAccount, SavingsAccount la fel in pachetul accounts, iar transaction si address le pastrez asa cum sunt?

diana-maftei replied:
asta e abordarea buna:
person(cu pojo, interfata si service strict pentru clasa Person)
employee(doar cu pojo-ul, intervata si service-ul pentru clasa Employee)

asta nu e abordarea buna:
Person, Customer si Employee cu toate pojo-urile si serviciile lor intr-un pachet persons

diana-maftei commented on project/src/com/company/accountservices/AccountServiceInterface.java:
```diff
@@ -0,0 +1,17 @@
+
+public interface AccountServiceInterface {
+
+    public void showAccounts();
```
[Improvement] cuvantul Account(s) este redundant in aceste metode, fiindca e clar la ce faci show, read, add, fiindca te afli in AccountService, si daca apelezi metoda din afara, o apelezi de pe un obiect accountService.

La fel si pentru celelalte clase, in loc sa ajute la citirea mai usoara a codului, o ingreuneaza

AmarKayed replied:
Intr-adevar, initial cand am denumit metodele crezusem ca are sens si ca asa o sa imi fie mai usor sa imi dau seama ce metode show pentru care clasa o sa apelez. Abia dupa mi-am dat seama ca oricum o sa fac apelurile de genul ClasaService.getInstance().metoda() si ca oricum apare numele clasei :)))
O sa remediez aceasta chestie imediat.

diana-maftei commented on project/src/com/company/accountservices/AccountServiceInterface.java:
```diff
@@ -0,0 +1,17 @@
+
+import com.company.accounts.Account;
+
+public interface AccountServiceInterface {
```
[Improvement] nu e nevoie sa adaugi Interface in nume, fiindca e clar in IDE dupa felul iconitei

diana-maftei commented on project/src/com/company/accountservices/CurrentAccountServiceInterface.java:
```diff
@@ -0,0 +1,23 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public interface CurrentAccountServiceInterface {
```
[Fix] ai putea extinde AccountServiceInterface, si doar sa adaugi metodele care nu exista deja acolo

AmarKayed replied:
Am vorbit pe Teams despre acest lucru si a ramas ca pot pastra AccountService si PersonService din moment ce ma folosesc de ele in serviciile claselor derivate. De asemenea, am ales sa nu extind CurrentAccountServiceInterface din AccountServiceInterface, intrucat CurrentAcountServiceInterface nu implementa toate metode care erau deja in AccountServiceInterface precum si faptul ca AccountServiceInterface se ocupa de stergerea generala a conturilor(fie ele de tip CurrentAccount sau SavingsAccount, clasa CurrentAccount a devenit abstracta dupa acest Pull Request si deci nu va avea obiecte). 

Source:
"Ma refer ca in loc sa scriu aceeasi citire a bazei Person in EmployeeService si CustomerService nu as putea sa las cum e acum si sa apelez direct citirea din PersonService chiar daca Person nu va avea instante? -> ok"

diana-maftei commented on project/src/com/company/accounts/Account.java:
```diff
@@ -0,0 +1,99 @@
+
+import java.util.Objects;
+
+public class Account {
```
[Fix] avand in vedere ca ai CurrentAccount si SavingsAccount, nu ar trebui aceasta clasa sa fie abstracta? poti crea un cont care sa nu fie unul din astea 2?

AmarKayed replied:
Solved: Am schimbat Account astfel incat sa fie abstract si am sters orice obiect intermediar de tip Account pe care l-as fi folosit pentru citirea obiectelor de tip CurrentAccount si SavingsAccount. Multumesc mult pentru observatie!

Commits:
https://github.com/AmarKayed/Advanced-OOP-Project/commit/81acc493a3904b924898e48eb7fc8dc40b9702be
^ Am schimbat Person in clasa abstracta
https://github.com/AmarKayed/Advanced-OOP-Project/commit/acc057aa0412555bd9b76b33607d721405635d0d
^ Am schimbat Account in clasa abstracta

diana-maftei commented on project/src/com/company/persons/Person.java:
```diff
@@ -0,0 +1,97 @@
+
+public class Person {
+
+    private static int nrOfPersons = 0;     // Used for Auto-Incrementing the Person.id attribute
```
[Fix] poti sa faci direct id-ul static si sa il incrementezi, nu ai nevoie de aceasta variabila in plus, si id-ul nu e nevoie sa fie final, doar sa nu aiba setter, fiindca odata ce a fost alocat un id in constructor, nu vei mai avea altfel cum sa il schimbi

AmarKayed replied:
Solved: Am schimbat doar id-ul cat sa nu mai fie final si am sters setter-ul pentru id. Id-ul static va ramane pentru incrementarea id-urilor.

diana-maftei commented on README.md:
```diff
@@ -1 +1,110 @@
 # This Is An Advanced Object Oriented Programming Project Written In Java!
+
+## **Description**
```
Foarte faina descriere, foarte detaliata, multumesc

AmarKayed replied:
Multumesc si eu! 😄

diana-maftei commented on project/src/com/company/Service.java:
```diff
@@ -0,0 +1,157 @@
+                CustomerService.getInstance().showCustomerList();
+                break;
+            case "B":
+                AccountService.getInstance().showAccounts();
```
[Improvement] sugerez sa pui toate aceste instante in niste variabile la inceputul clasei, nu vrei sa apelezi getInstance pe toate aceste servicii de fiecare data, e un extra method call care iti ia milisecunde in plus la rularea aplicatiei, plus ca ingreuneaza citirea

AmarKayed replied:
Am facut asta alocandu-mi niste atribute finale in clasa Service. M-am gandit ca nu ar avea sens sa fac aceasta modificare si in restul claselor din moment ce doar in Service o sa apelez toate serviciile create.
https://github.com/AmarKayed/Advanced-OOP-Project/commit/1be296a847ddbcb2c9a07f92e75f243b006ffba4

diana-maftei commented on project/src/com/company/accounts/Account.java:
```diff
@@ -0,0 +1,99 @@
+        return IBAN;
+    }
+
+//    public void setIBAN(int IBAN) {
```
[Improvement] il poti scoate cu totul, nu doar comenta

AmarKayed replied:
Acum observ ca am pastrat anumite bucati de cod pe care nu le-am mai folosit comentate. O sa le sterg astfel incat sa fie codul mai putin incarcat.

diana-maftei commented on project/src/com/company/accounts/CurrentAccount.java:
```diff
@@ -0,0 +1,81 @@
+
+    public CurrentAccount(String openDate, Customer holder, float initialBalance, float commission, List<Transaction> transactionHistory) {
+        super(openDate, holder, initialBalance);
+        this.commission = commission;
```
[Fix] ai multi constructori care seteaza acelasi lucru si cred ca poti simplifica apelandu-se unul pe altul

AmarKayed replied:
Aveti dreptate, am rezolvat aceasta problema prin a sterge majoritatea constructorilor. Initial am ales sa rescriu majoritatea constructorilor cat sa se apeleze intre ei acolo unde se poate dar dupa mi-am dat seama ca:
1. Am multe warning-uri pe IntelliJ, intrucat nu foloseam toti acei constructori cam nicaieri
2. Nu prea aveam nevoie de acei constructori

It was a sudden and tough breakup but it was for the greater good. :")

diana-maftei commented on project/src/com/company/accounts/CurrentAccount.java:
```diff
@@ -0,0 +1,81 @@
+    }
+
+    public CurrentAccount(Account ob, float commission) {
+        super(ob);
```
aici, spre exemplu, ai putea avea o singura linie de cod, care apeleaza this(ob, commission, new ArrayList<>())

AmarKayed replied:
Intr-adevar acum observ cat de redundant am facut multe chestii, honestly m-a luat putin valul cu functia de generate de pe IntelliJ :)))))

Solved: 
https://github.com/AmarKayed/Advanced-OOP-Project/pull/1/files#r841170699

diana-maftei commented on project/src/com/company/transaction/Transaction.java:
```diff
@@ -0,0 +1,110 @@
+
+import java.util.*;
+
+public class Transaction implements Comparable{
```
[Fix] nu folosi Comparable raw, foloseste-l cu generice:  Comparable<Transaction>, vezi ca si Intellij se plange
![image](https://user-images.githubusercontent.com/67580205/161415802-ec6e4bba-391f-4ba3-92b7-18f44b4a2991.png)
Oricand vezi in intelli un cuvant highlightuit asa, sa faci hover findca sigur iti ofera o sugestie de imbunatatire

AmarKayed replied:
Am rezolvat acest lucru prin a schimba Comparable in Comparable<Transaction>.

Commit: 
https://github.com/AmarKayed/Advanced-OOP-Project/commit/6cf64731dc73781b1a5dbc6d79d3ed097bf12b09

diana-maftei commented on project/src/com/company/persons/Person.java:
```diff
@@ -0,0 +1,97 @@
+
+import java.util.Objects;
+
+public class Person {
```
[Fix] de ce nu e aceasta clasa abstracta? poti crea obiecte de tipul Person care sa nu fie Customer sau Employee? eu nu cred ca ar trebui sa iti permita sistemul sa faci asta

AmarKayed replied:
Este o observatie chiar foarte buna. Nu mi-a trecut prin cap sincer, aveti dreptate. 
Solved: Am schimbat Person astfel incat sa fie abstract si am sters orice obiect intermediar de tip Person pe care l-as fi folosit pentru citirea obiectelor de tip Employee sau Customer.

diana-maftei commented on project/src/com/company/personservices/PersonService.java:
```diff
@@ -0,0 +1,181 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class PersonService implements PersonServiceInterface{
```
[Fix] cred ca ar trebui sa renunti la acest serviciu cu totul deoarece Person ar trebui sa fie abstract, si nu ar trebui sa iti permita sa creezi sau sa modifici persoane

AmarKayed replied:
Am vorbit pe Teams despre acest lucru si a ramas ca pot pastra AccountService si PersonService din moment ce ma folosesc de ele in serviciile claselor derivate. Am transformat clasele Person si Account in clase abstracte in urma acestul Pull Request, insa am ales sa pastrez serviciile acestor clase ca baza pentru serviciile claselor derivate(de exemplu pentru citirea unui Employee sau Customer ma voi folosi de aceeasi citire din PersonService, etc).

Source:
"Ma refer ca in loc sa scriu aceeasi citire a bazei Person in EmployeeService si CustomerService nu as putea sa las cum e acum si sa apelez direct citirea din PersonService chiar daca Person nu va avea instante? -> ok"

diana-maftei commented on project/src/com/company/accountservices/AccountService.java:
```diff
@@ -0,0 +1,146 @@
+import java.util.List;
+import java.util.Map;
+
+public class AccountService implements AccountServiceInterface{
```
[Fix] nu cred ca ar trebui sa ai aceasta clasa, fiindca Account ar trebui sa fie abstract, si sa ai doar implementari pentru tipuri de conturi current si savings

AmarKayed replied:
Am vorbit pe Teams despre acest lucru si a ramas ca pot pastra AccountService si PersonService din moment ce ma folosesc de ele in serviciile claselor derivate.

Source:
"Ma refer ca in loc sa scriu aceeasi citire a bazei Person in EmployeeService si CustomerService nu as putea sa las cum e acum si sa apelez direct citirea din PersonService chiar daca Person nu va avea instante? -> ok"

diana-maftei commented on project/src/com/company/Main.java:
```diff
@@ -0,0 +1,15 @@
+public class Main {
+
+
+    public static void main(String[] args) {
```
![image](https://user-images.githubusercontent.com/67580205/161416169-968bd348-6c9e-44b5-80d6-7114ab88552e.png)
se comporta ciudat daca nu exista deja un cont. ori ascunde optiunea aceea, daca nu exista un cont, ori afiseaza un mesaj corespunzator
![image](https://user-images.githubusercontent.com/67580205/161416251-9a007c4b-71f1-49fb-a1d5-a530a38b9e44.png)
nu te astepta ca utilizatorii sa foloseasca corect aplicatia, multi sunt complet paralel cu tehnologia si o sa bage tot felul de aberatii

AmarKayed replied:
Update: Momentan am rezolvat doar bug-urile in care incercam sa folosim o comanda ce are nevoie de date, atunci cand nu exista date introduse in aplicatie. Urmeaza sa rezolv si problemele cand se tasteaza un string atunci cand se cere un numar si viceversa.

AmarKayed committed:
Pull Request Fix: Renamed "IBAN" to "iban"

I used the refractor option in IntelliJ.

AmarKayed committed:
Pull Request Fix: Started Working On Fixing Input Problems

The problem is that the app crashes when we try to open a bank account without first having any customers or trying to add transactions when we don't have any bank accounts open.

AmarKayed committed:
Pull Request Fix: Finished Fixing Input Problems

I revised every single Service command available and tried to use it without entering any data and then fixed the issues.

AmarKayed committed:
Pull Request Fix: Fixed Final Person ID + Transformed Person Class To Abstract Class

Changing the Person class to an abstract class required changing the behaviour of the PersonService.readPerson() method, as well as the ways in which we read Employees and Customers, since they derive from the Person class.

AmarKayed committed:
Pull Request Fix: Fixed ID Assignment Issue On Derived Class Instance Creation

The problem was with the "removeTemporaryObject()" methods, since in the last commit we've transformed the base classes to abstract classes, there were no more temporary base class objects created(abstract classes donot allow instances to be created). In this commit we've removed that method entirely since we're not going to be needing it anymore.

AmarKayed committed:
Pull Request Imporvement: Replaced All "getInstance()" Calls With Instance Variables

This change will have lasting effect on the readability of the code. Basically, all I've done is replace all the "getInstance()" method calls in the "Service.run()" method with variables of those instances such that it is shorter and more readable.

AmarKayed committed:
Pull Request Fix: Removed Unused Constructors

Many of the constructors either could've been written otherwise or weren't ever actually used and thus were rewritten or removed.

AmarKayed committed:
Pull Request Fix: Replaced "Comparable" With "Comparable<Transaction>" In Transaction Class

The problem was that I shouldn't have extended the raw "Comparable" class. Instead, I should've given it a type which coincides with the class for which I'm implementing the "compareTo()" method.

AmarKayed committed:
Pull Request Fix: Fixed Input Bugs

Starting from this commit, the application will have no problem with intaking strings when it requires numbers as input. The only problem that remains is when an invalid date is typed. My guess is that I can solve this either by using regular expressions or by implmenting a Date class.

AmarKayed commented on project/src/com/company/Main.java:
Am rezolvat toate problemele de input(cel putin cele pe care le-am gasit). 
Singura problema de input care ramane ar fi atunci cand se tasteaza o data cu format invalid(care sa nu fie ZZ LUNA AA).
Aceasta problema apare doar atunci cand incercam sa sortam in functie de data. In rest, se poate primi ca input(care in sine e o mare problema).
Solutia ar fi sa ma folosesc fie de Regular Expressions ca sa detectez daca formatul tasatat este valid fie(intr-un caz mai complicat) sa imi implementez propria clasa Date care sa se ocupe cu input-ul pentru date calendaristice(pentru format-ul si context-ul acestei aplicatii).

Commit:
https://github.com/AmarKayed/Advanced-OOP-Project/commit/13492375569df799cfa5e9ea9ba50ff41cfe356f

AmarKayed committed:
Pull Request Improvement: Refractored Service Classes

I basically changed all the names of the Service class implementations by adding an "Impl" at the end and removed "Interface" from the Service class interfaces.

AmarKayed commented on project/src/com/company/accountservices/AccountServiceInterface.java:
Am modificat numele claselor de servicii si ale interfetelor acestora, asa cum am discutat si pe Teams.

Source:
"legat de nume, de obicei vei avea interfata cu numele serviciului si implementarea va fi AccountServiceImpl"
"adica implementarea e cea care are numele diferit"

Commit:
https://github.com/AmarKayed/Advanced-OOP-Project/commit/c52e9f47c436b1c81c60a874cbafa0cb458b9f19

AmarKayed committed:
Pull Request Fix: Refractored Package Folders

I basically redesigned the whole folder structure of the project such that the POJO, Service and Service Interface files are layered by business logic.

AmarKayed commented on project/src/com/company/accounts/Account.java:
Business Logic Layers it is!

Am am ales sa reorganizez structura pachetelor dupa BLL.

Commit:
https://github.com/AmarKayed/Advanced-OOP-Project/commit/5c6dd1967888208eb6a0af7d5290761e05292761

AmarKayed committed:
Pull Request Fix: Removed Draft Files

I chose to remove the draft files before merging with the main branch, instead of using git stash.

AmarKayed commented on project/src/com/company/draftignore/DraftClassIgnore.java:
Am ales sa le sterg cu totul de pe git, intrucat nu prea aveam nevoie de ele oricum decat pentru teste. Multumesc pentru link, o sa incerc sa folosesc git stash pe viitor.

Commit:
https://github.com/AmarKayed/Advanced-OOP-Project/commit/6978b8ad0f8fe01f0bf3fb71f5ab80a43574b8de

AmarKayed committed:
Pull Request Improvement: Removed Commented Code + Unused Imports

I'm not sure if this was an actual Pull Request Improvement but I did it anyway for a cleaner code.

AmarKayed committed:
Pull Request Improvement: Refractored Show, Read, Add, Update, Delete Methods

I basically simplified the names of these methods such that for example for the AccountService class we don't call "AccountService.showAccounts()" since we already know we're talking about accounts. This change makes the code more readable and somewhat lighter on the eyes.

AmarKayed commented on project/src/com/company/accountservices/AccountServiceInterface.java:
Am facut cum ati sugerat si simt ca am dus codul la frizer :))))). Este mult mai lizibil acum mai ales ca am aplicat si tips-ul legat de getInstance(). Multumesc mult!

Commit:
https://github.com/AmarKayed/Advanced-OOP-Project/commit/70bba8d2027de37bf063de140104ca01dc7cf6f1
